lightly/api/prediction_singletons.py

```diff
@@ -0,0 +1,24 @@
+from abc import ABC
+from typing import Optional, List
+
+from lightly.openapi_generated.swagger_client import TaskType
+
+
+class PredictionSingletonRepr(ABC):
+    type: str
+    taskName: str
+    cropDatasetId: Optional[str] = None
+    cropSampleId: Optional[str] = None
+    categoryId: int
+    score: float
```

Comment:
Is the reason not to use dataclasses that we support python 3.6?

Reply:
I would recommend using instance variables instead of class variables; that way we can prevent sharing the instances. Without dataclasses there is a little boilerplate code:
```py
class PredictionSingletonRepr(ABC):
    def __init__(...)
        self.type = type
        self.taskName = taskName
        self.categoryId = categoryId
        self.score = score

class PredictionSingletonClassificationRepr(PredictionSingletonRepr):
    def __init__(...):
        super().__init__(...)
        self.probabilities = probabilities
```

Reply:
> Is the reason not to use dataclasses that we support python 3.6?

That and we need custom initialisers.